src/pocketmine/permission/BanEntry.php

```diff
@@ -82,43 +82,38 @@ public function setReason($reason){
 		$this->reason = $reason;
 	}
 
-	public function getString(){
-		$str = "";
-		$str .= $this->getName();
-		$str .= "|";
-		$str .= $this->getCreated()->format(self::$format);
-		$str .= "|";
-		$str .= $this->getSource();
-		$str .= "|";
-		$str .= $this->getExpires() === null ? "Forever" : $this->getExpires()->format(self::$format);
-		$str .= "|";
-		$str .= $this->getReason();
-
-		return $str;
+	public function getArray(){
+		$data = [];
+		$data[] = $this->getName();
+		$data[] = $this->getCreated()->format(self::$format);
+		$data[] = $this->getSource();
+		$data[] = $this->getExpires() === null ? "Forever" : $this->getExpires()->format(self::$format);
+		$data[] = $this->getReason();
+
+		return $data;
 	}
 
 	/**
-	 * @param string $str
+	 * @param array $data
 	 *
 	 * @return BanEntry
 	 */
-	public static function fromString($str){
```

Comment:
This is not the proper way to deprecate a method. You should remain basic support for the method, and mark it as deprecated.